provider/gce/environ_broker.go

```diff
@@ -131,6 +131,27 @@ func (env *environ) findInstanceSpec(
 	return spec, errors.Trace(err)
 }
 
+func (env *environ) imageURLBase(os jujuos.OSType) (string, error) {
+	base, useCustomPath := env.ecfg.baseImagePath()
+	if useCustomPath {
+		return base, nil
+	}
+
+	if os == jujuos.Ubuntu {
+		base = ubuntuImageBasePath
+	} else if os == jujuos.Windows {
+		return windowsImageBasePath, nil
+	} else {
+		return "", errors.Errorf("os %s is not supported on the gce provider", os.String())
+	}
+
+	if env.Config().ImageStream() == "daily" {
```

Comment:
This is a bit messy - let's redo the if then else above as a switch and have all os related logic under ubuntu or windows

Reply:
Fixed :hammer: